fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulkcommon/BulkDataUtils.java

```diff
@@ -145,101 +157,159 @@ private static int getFhirResourceFromBufferReader(BufferedReader resReader, int
                 if (lineRed <= numOfLinesToSkip) {
                     continue;
                 }
-                fhirResources.add(FHIRParser.parser(Format.JSON).parse(new StringReader(resLine)));
-                exported++;
-                if (exported == Constants.IMPORT_NUMOFFHIRRESOURCES_PERREAD) {
-                    break;
+                try {
+                    fhirResources.add(FHIRParser.parser(Format.JSON).parse(new StringReader(resLine)));
+                    exported++;
+                    if (exported == Constants.IMPORT_NUMOFFHIRRESOURCES_PERREAD) {
+                        break;
+                    }
+                } catch (FHIRParserException e) {
+                    // Log and skip the invalid FHIR resource.
+                    logger.warning("getFhirResourceFromBufferReader: " + e.getMessage());
+                    logger.warning("getFhirResourceFromBufferReader: " + "Failed to parse '" + resLine + "'");
+                    parseFailures++;
+                    continue;
                 }
             }
         } while (resLine != null);
-        return exported;
+        return parseFailures;
```

Comment:
why would export result in parse failures? wouldnt it already be valid and stored in the db

also why is it reporting back the number of parseFailures? maybe update the method signature to explain the contract and purpose

Reply:
this is the read step, the success ones are in the fhirresource List, so can easily get the number from the list using size(), failure number is needed to update the data source for the metrics in the write step later.

Reply:
got it